### DISCUSSION TITLE
Future with React Server Component

### DISCUSSION THREAD

**Discussion by ashvin777** [remix-run/remix#898] [General]
I am very much interested in using Remix Run for building my next project, however I am planning to wait for React 18 to be launched with the most awaited feature (React Server Component). 

I would like to know if Remix will be interested in supporting RSC or not. Next.js released beta with RSC already, so any plan for Remix to support the same?

**Comment by MichaelDeBoey** [comment]
I know the team already said in the Discord server that they're planning to support Server Components at some point.

But with Remix the benefits they give you won't actually be needed (in most cases).

**Reply by ashvin777** [reply]
I would like to know more about why it wont be needed, do we have similar pattern solving the same problem ?

**Reply by muraterzioglu** [reply]
I think this video will be helpful; 
[Remix is a NEW JavaScript framework you MUST try](https://www.youtube.com/watch?v=r4B69HAOXnA) by Fireship

**Reply by sergiodxa** [reply]
@ashvin777 because in Remix your data is tied to the route instead of individual components, and Remix can already get the data of every route in parallel, there's too little benefit of using RSC, Remix may eventually use it internally, but it's something you most likely are not going to need to care about and get the benefit of shipping less JS automatically once you upgrade.

**Reply by ashvin777** [reply]
The much anticipation is for the zero bundle size that will reduce the bundle size a lot.

Does Remix uses similar techinque to reduce the bundle size? One of the painfull module is react-dom itself, it needs to be shipped for each route for hydration. I dont think in RSC we do that.

**Reply by sergiodxa** [reply]
@ashvin777 right now in Remix you can remove the Scripts component and avoid JS directly if you want/need, and you can enabled/disable it in a route basis, so some routes may live without JS and if you go to a more dynamic route it can load JS.

**Reply by ashvin777** [reply]
Ok I understand, basically you are talking about skipping hydration. However for interactivity, we would need to ship the bundle same as Next.js/SSR.

**Comment by MichaelDeBoey** [comment]
@ryanflorence wrote the official stand of the @remix-run team on this topic

https://remix.run/blog/react-server-components

**Reply by ashvin777** [reply]
Awesome, thanks for sharing!
